### PR TITLE
Eliegab21 feature s3 eventbridge gluecrawler

### PR DESCRIPTION
**Pull request by ElieGab21**
*Issue #, if available:*

*Description of changes:*

This pull request is for the submission of a new pattern for Serverless land. This pattern creates a State Machine triggered by an S3 notification. The state machine triggers a glue crawler and a glue ETL job to transform and catalog csv data. It creates all the necessary permissions and roles, following the principle of least privilege. 

The changes add a new folder to the repository called s3-eventbridge-glue, containing all the folder structure of a Python CDK project as well as fake sample data to test the pattern with. The main elements are the stack: s3_eventbridge_gluecrawler_stack.py and the requirements.txt file containing the required packages to install for the pattern to deploy correctly. Finally, it includes a README file containing the instructions to deploy, test and delete in the stack, as well as all the necessary information to understand what is being deployed.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: s3-eventbridge-glue/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+1. Starts a AWS Glue Crawler to crawl the input bucket. This allows you to have the raw data (before transformation) inside the Data Catalog
+2. Start a AWS Glue ETL Job, that runs a simple transformation job and drops empty columns in the dataset
+
+The processed/transformed data is then added to the Data Catalog and to an S3 bucket to allow for further processing.
+
+## Testing
+
+1. Navigate to the S3 bucket containing 'input-bucket' in the name
+2. Upload a .csv file to the bucket. A [sample file](./sample_data/sample_dataset.csv) can be found in this pattern.
+3. Wait for the AWS Step Functions State Machine to complete and you should be able to see 2 tables under the Glue Data Catalog, one containing the raw and the other the processed data.
```
Comment (pputhran) [inline s3-eventbridge-glue/README.md:70]: can you add a screenshot of what the end result should look like?

File: s3-eventbridge-glue/example-pattern.json [hunk 2]
```diff
@@ -0,0 +1,63 @@
+{
+    "title": "Automatic Data Transformation and Catalog with Amazon S3 notifications and AWS Step functions",
```
Comment (pputhran) [inline s3-eventbridge-glue/example-pattern.json:2]: `title`: title does not meet maximum length of 75.

Can you shorten the title and also include Glue in there as well?

File: s3-eventbridge-glue/example-pattern.json [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    "description": "This pattern facilitates the cataloging and transformation of CSV data, using an event-driven architecture to automatically trigger data cleaning and transformation jobs",
```
Comment (pputhran) [inline s3-eventbridge-glue/example-pattern.json:3]: `description`: description does not meet maximum length of 150

File: s3-eventbridge-glue/example-pattern.json [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        "When you upload a .csv file to the Input S3 bucket, it matches with an Amazon EventBridge rule that triggers a State Machine. The State Machine is composed of two steps:",
+        "1. Starts a AWS Glue Crawler to crawl the input bucket. This allows you to have the raw data (before transformation) inside the Data Catalog",
+        "2. Start a AWS Glue ETL Job, that runs a simple transformation job and drops empty columns in the dataset" ,
+        "The processed/transformed data is then added to the Data Catalog and to an S3 bucket to allow for further processing."
+      ]
+    },
+    "gitHub": {
+      "template": {
+        "repoURL": "https://github.com/aws-samples/serverless-patterns/tree/main/s3-eventbridge-glue",
+        "templateURL": "serverless-patterns/s3-eventbridge-glue",
```
Comment (pputhran) [inline s3-eventbridge-glue/example-pattern.json:19]: `gitHub.template.templateURL`: gitHub.template.templateURL Please remove the projectFolder value from the templateFile

File: s3-eventbridge-gluecrawler/README.md [hunk 5]
```diff
@@ -0,0 +1,60 @@
+# AWS Service 1 to AWS Service 2
```
Comment (pputhran) [inline s3-eventbridge-gluecrawler/README.md:1]: Are these files under the `s3-eventbridge-gluecrawler` folder added in error? Can you please review these?

**Review by pputhran — CHANGES_REQUESTED** [review]
Have added a few comments

File: s3-eventbridge-gluecrawler/README.md (hunk 5 above)
Comment (ElieGab21) [reply s3-eventbridge-gluecrawler/README.md]: Resolved - sorry it was my mistake

File: s3-eventbridge-glue/example-pattern.json (hunk 2 above)
Comment (ElieGab21) [reply s3-eventbridge-glue/example-pattern.json]: Resolved

File: s3-eventbridge-glue/example-pattern.json (hunk 3 above)
Comment (ElieGab21) [reply s3-eventbridge-glue/example-pattern.json]: Resolved

File: s3-eventbridge-glue/example-pattern.json (hunk 4 above)
Comment (ElieGab21) [reply s3-eventbridge-glue/example-pattern.json]: Not sure what you mean by that?

Comment (ElieGab21) [reply s3-eventbridge-glue/example-pattern.json]: Resolved

File: s3-eventbridge-glue/README.md (hunk 1 above)
Comment (ElieGab21) [reply s3-eventbridge-glue/README.md]: Resolved

File: s3-eventbridge-glue/example-pattern.json [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        "templateFile": "s3-eventbridge-gluecrawler/s3-eventbridge-gluecrawler.py"
```
Comment (pputhran) [inline s3-eventbridge-glue/example-pattern.json:20]: - gitHub.template requires the property "projectFolder". 
- Can you rename the folder `s3-eventbridge-gluecrawler` to `s3-eventbridge-crawler` and then update the `templateFile` accordingly that should all the issues.

**Review by pputhran — CHANGES_REQUESTED** [review]

File: s3-eventbridge-glue/example-pattern.json (hunk 6 above)
Comment (ElieGab21) [reply s3-eventbridge-glue/example-pattern.json]: Resolved

**Review by pputhran — APPROVED** [review]

**Pull request merged**